### PR TITLE
Docmosis http to https redirect

### PR DESCRIPTION
**Pull request by hannah38**
Add listener, redirect config and redirect rule for http to https redirection in certain cases including docmosis.
Testing with set to true on docmosis sandbox, loop with a conditional to enable this http-https redirection.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: environments/sbox/backend_lb_config.yaml [hunk 1]
```diff
@@ -93,4 +93,5 @@ gateways:
         component: docmosis
         cookie_based_affinity: Enabled
         ssl_enabled: true
-        host_name_prefix: docmosis
\ No newline at end of file
+        host_name_prefix: docmosis
+        http_to_https_redirect: true
```
Comment (timja) [inline environments/sbox/backend_lb_config.yaml:97]: is this just for like BA's accessing the web ui?

Comment (hannah38) [reply environments/sbox/backend_lb_config.yaml:97]: Yeah, just for UI - after login it goes to http and a 404

Comment (timja) [reply environments/sbox/backend_lb_config.yaml:97]: that's masking an issue then, it shouldn't be sending you to http, either x-forwarded-* headers are missing (unlikely but check), or something needs to be done at docmosis to tell it that its behind a proxy

Comment (adusumillipraveen) [reply environments/sbox/backend_lb_config.yaml:97]: tried few things and couldn't find anything specific and makes it difficult as we are on a old version of Docmosis.

**Pull request closed without merging**